--- functions.php.original	2022-02-18 23:07:04.661123582 -0600
+++ functions.php	2022-02-18 23:30:20.489093086 -0600
@@ -267,6 +267,19 @@
 	/** Get server name from input */
 	$server_location = getServerPath($serverhost);
 	extract($server_location);
+
+    /** URL/Hostname facileManager will use to talk to this client **/
+	echo "Please enter the server name of the $module_name module:\n";
+	echo "    Examples incldue:\n";
+	echo "\t$module_name\n";
+	echo "\t$module_name.mydomain.com\n";
+	echo "Please enter the server name of the $module_name module: ";
+	if (defined('HOST')) {
+		$data['server_name'] = HOST;
+		echo HOST . "\n";
+	} else {
+		$data['server_name'] = trim(fgets(STDIN));
+	}
 	
 	$data['config'] = array();
 
@@ -296,6 +309,8 @@
 	}
 	
 	$data['config'][] = array('FMHOST', 'facileManager server', $hostname . '/' . $path);
+
+	$data['config'][] = array('HOST', 'facileManager module url', $data['server_name']);
 	
 	/** Account key **/
 	$key = 'default';
@@ -322,8 +337,6 @@
 	}
 
 	/** Server serial number **/
-	$data['server_name'] = gethostname();
-
 	$data['server_os'] = PHP_OS;
 	$data['server_os_distro'] = detectOSDistro();
 	echo fM('Please enter the serial number for ' . $data['server_name'] . ' (or leave blank to create new): ');
@@ -673,6 +686,7 @@
 		$distros = array(
 			'Arch'       => '/etc/arch-release',
 			'ALT'        => '/etc/altlinux-release',
+			'Debian'     => '/etc/debian_version',
 			'Sabayon'    => '/etc/sabayon-release',
 			'Slackware'  => '/etc/slackware-version;/etc/slackware-release',
 			'SUSE'       => '/etc/SuSE-release;/etc/UnitedLinux-release',
